cpp/cmake/Modules/EvalGPUArchs.cmake

```diff
@@ -14,12 +14,21 @@
 # limitations under the License.
 #=============================================================================
 
+# Unset this first in case it's set to <empty_string>
+unset(CMAKE_CUDA_ARCHITECTURES CACHE)
+
+# Enable CUDA so we can invoke nvcc
+enable_language(CUDA)
+
+# Function uses the CUDA runtime API to query the compute capability of the device, so if a user
+# doesn't pass any architecture options to CMake we only build the current architecture
 function(evaluate_gpu_archs gpu_archs)
   set(eval_file ${PROJECT_BINARY_DIR}/eval_gpu_archs.cu)
   set(eval_exe ${PROJECT_BINARY_DIR}/eval_gpu_archs)
   set(error_file ${PROJECT_BINARY_DIR}/eval_gpu_archs.stderr.log)
-  file(WRITE ${eval_file}
-[=[
+  file(
+    WRITE ${eval_file}
+    "
```

Comment:
Yes to align with RMM's behavior, specifically [this line](https://github.com/rapidsai/rmm/blob/branch-0.19/cmake/Modules/EvalGPUArchs.cmake#L45) returning the list of supported archs instead of "ALL"